src/js/components/TweetTextField.js

```diff
@@ -21,25 +21,36 @@ export function TweetTextField() {
 		return siteUrl.length;
 	};
 
-	const overrideLengthClass = () => {
-		if ( 280 <= permalinkLength + tweetText.length ) {
-			return 'over-limit';
+	const { permalinkLength, maxLength } = useSelect( ( select ) => {
+		return {
+			permalinkLength: getPermalinkLength( select ),
+			maxLength: ( 275 - getPermalinkLength( select ) ),
+		};
+	} );
+
+	const getTweetLength = () => {
+		// +5 because of the space between body and URL and the ellipsis.
+		const length = permalinkLength + tweetText.length + 5;
+		if ( 280 <=  length ) {
+			return { tweetLength: sprintf( __( '%d - Too Long!', 'autoshare-for-twitter' ), length ), overrideLengthClass: 'over-limit' };
```

Comment:
@iamdharmesh Can document `%d` in `translator comment`. It is helpful in translation plugins.